File: leftRotation.py
# https://www.hackerrank.com/challenges/ctci-array-left-rotation/problem


def rotLeft(a, d):
    return a[d:] + a[:d]


# tests
test_a = [1, 2, 3, 4, 5]
test_d = 4
expected_output = [5, 1, 2, 3, 4]
print('Passes:', rotLeft(test_a, test_d) == expected_output)

File: hackerrank/leftRotation.js
// https://www.hackerrank.com/challenges/ctci-array-left-rotation/problem

function rotLeft(a, d) {
  return [...a.slice(d), ...a.slice(0, d)];
}

// tests
const test_a = [1, 2, 3, 4, 5];
const test_d = 4;
const expected_output = [5, 1, 2, 3, 4];
console.log(
  'Passes:',
  rotLeft(test_a, test_d).toString() === expected_output.toString()
);
